src/util/records.py

```diff
@@ -1479,7 +1479,15 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
             continue
         if ir.company_dependent_comodel:
             if ir.company_dependent_comodel == model_src:
-                assert model_src == model_dst
+                if model_src != model_dst:
+                    cr.execute(
+                        "UPDATE ir_model_fields SET relation = %s WHERE model = %s AND name = %s",
+                        (
+                            model_dst,
+                            model_of_table(cr, ir.table),
+                            ir.res_id,
+                        ),
```

Comment:
nit: IMHO it helps readability by reducing chances of confusing with a function call. Feel free to ignore it.
```suggestion
                        [
                            model_dst,
                            model_of_table(cr, ir.table),
                            ir.res_id,
                        ],
```